app/models/active_operator/operation.rb

```diff
@@ -1,15 +1,20 @@
 class ActiveOperator::Operation < ApplicationRecord
   belongs_to :record, polymorphic: true
 
-  def operator
-    operator_class.new(self)
-  end
-
-  delegate :request, :process, to: :operator
+  def perform
+    save! unless persisted?
 
-  def operate!
     request!
     process!
+  rescue
+    errored!
+    raise
+  end
+
+  def perform_later
+    save! unless persisted?
```

Comment:
I wonder if it'd be clearer if we used `create_#{name}` in `has_operation` and spared the record having to keep concerning itself with whether it's persisted.

Reply:
That's interesting...

Doing that and calling `Location.new.geocoding` raises `You cannot call create unless the parent is saved (ActiveRecord::RecordNotSaved)`. Does that seem surprising? Maybe it's a good protection against reaching into the operation prior to persistence?

Reply:
Another thing, having fewer conditions in the operations makes it simpler to reason about from the user's perspective. And if this is to be a gem, which they're meant to rely on, having fewer states that they need to keep in mind when using it is better. It's small in this case, but still.

And having an `ActiveRecord::RecordNotFound` exception thrown helps communicate that they're in an unexpected state. You could then document that in the README.

I think this is also consistent with ActiveJob works if you pass it an unpersisted record.

Reply:
Just double-checked with Active Job, it'll raise:

```
ActiveJob::SerializationError: Unable to serialize Post::Publisher without an id. (Maybe you forgot to call save?)
```

Reply:
>And if this is to be a gem, which they're meant to rely on, having fewer states that they need to keep in mind when using it is better. It's small in this case, but still.

Ah, that makes sense!

>I think this is also consistent with ActiveJob works if you pass it an unpersisted record.

Cool, I think you've convinced me, thanks for the perspective! 🙏